cli/src/main/java/hudson/cli/CLI.java

```diff
@@ -456,18 +522,35 @@ public boolean verify(String s, SSLSession sslSession) {
                 sshAuthRequestedExplicitly = true;
                 continue;
             }
+            if (head.equals("-strictHostKey")) {
+                strictHostKey = true;
+                args = args.subList(1, args.size());
+                continue;
+            }
+            if (head.equals("-user") && args.size() >= 2) {
+                user = args.get(1);
+                args = args.subList(2, args.size());
+                continue;
+            }
+            if (head.equals("-auth") && args.size() >= 2) {
+                auth = args.get(1);
+                args = args.subList(2, args.size());
+                continue;
+            }
             if(head.equals("-p") && args.size()>=2) {
                 httpProxy = args.get(1);
                 args = args.subList(2,args.size());
                 continue;
             }
-            if(head.equals("-v")) {
-                args = args.subList(1,args.size());
-                Logger l = Logger.getLogger(ROOT_LOGGER_NAME);
-                l.setLevel(FINEST);
-                for (Handler h : l.getHandlers()) {
-                    h.setLevel(FINEST);
+            if (head.equals("-logger") && args.size() >= 2) {
+                Level level = parse(args.get(1));
+                ConsoleHandler h = new ConsoleHandler();
+                h.setLevel(level);
+                for (Logger logger : new Logger[] {LOGGER, PlainCLIProtocol.LOGGER, Logger.getLogger("org.apache.sshd")}) { // perhaps also Channel
+                    logger.setLevel(level);
+                    logger.addHandler(h);
```

Comment:
Actually it does, though only for messages at `INFO` and above (if there are any at all), which matters little when you are seeing all kinds of messages at `FINE` or whatever you set the level to. However if you do not attach a new `Handler` then `-logger` will have no effect at all, since the default handler is set to `INFO`.

@ydubreuil’s trick from #2711 solves the conflict, so I will use that. As a bonus, it also makes `-logger OFF` and the like work.